sat_vapor_pres/include/sat_vapor_pres_k.inc

```diff
@@ -326,86 +176,98 @@
 ! differencing des values in the table
 
      do i = 2, table_size-1
-       D2TABLE3(i) = 0.25*dtinvl*(DTABLE3(i+1)-DTABLE3(i-1))
+       D2TABLE3(i) = 0.25_r8_kind*dtinvl*(DTABLE3(i+1)-DTABLE3(i-1))
      enddo
 ! one-sided derivatives at boundaries
 
-     D2TABLE3(1) = 0.50*dtinvl*(DTABLE3(2)-DTABLE3(1))
+     D2TABLE3(1) = 0.50_r8_kind*dtinvl*(DTABLE3(2)-DTABLE3(1))
 
-     D2TABLE3(table_size) = 0.50*dtinvl*&
-          (DTABLE3(table_size)-DTABLE3(table_size-1))
+     D2TABLE3(table_size) = 0.50_r8_kind*dtinvl*(DTABLE3(table_size)-DTABLE3(table_size-1))
    endif
 
       use_exact_qs = use_exact_qs_input
       module_is_initialized = .true.
 
- end subroutine sat_vapor_pres_init_k
+  end subroutine SAT_VAPOR_PRES_INIT_K_
 
 !#######################################################################
 
- function compute_es_k(tem, TFREEZE) result (es)
- real, intent(in) :: tem(:), TFREEZE
- real :: es(size(tem,1))
+ function COMPUTE_ES_K_(tem, TFREEZE) result (es)
```

Comment:
These routines have a lot of constants which are being cast up to real(8).  It might be worth looking into making these constants more accurate, and casting them down to real(4).